ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -519,7 +612,7 @@ class WebSocketService(
                 shiftedPrefix,
                 Terminates.Never,
               )
-              .via(convertFilterContracts(fn))
+              .via(convertFilterContracts(txsPred.fn))
```

Comment:
Two issues I see here, though I cannot quite say what is the best way to fix them:

1. `shiftedPrefix` is determined by `prefilter`, and if an ACS update was made, it is the offset that the ACS update reached (hence, "shifted" from the original offset passed into the function). It comes from `fetchAndPersist`.  As you see from the following, an ACS-and-txns stream is the concatenation of the prefiltered (ACS + live-begin) and liveFiltered (txn) parts. So, when an ACS update occurs, you are *starting* the live-filtered portion at whatever offset the ACS update ended at, which will likely be *after* whatever offsets were passed in.
2. If a query has an ACS portion, its function-predicate is *still* required during the liveFiltered portion, but your partition in `splitRequest` excludes the function-predicates for any query with an ACS portion from the predicate passed on line 615 to `convertFilterContracts`, *unless* there happen to be zero ACS-less queries.